### DISCUSSION TITLE
Docker network_mode: host - Why?

### DISCUSSION THREAD

**Discussion by AlbertoKarakoutev** [music-assistant/support#2734] [Q&A] [answered]
Could someone explain WHY this parameter is required? My docker compose config requires a network being setup with some network specific params, which I cant apply with the `network_mode: host` parameter. I want my containers to be logically separated from the host network and accessible to one another. HASS is running on my custom network and I want Music Assistant to do so as well. If someone could elaborate what exactly `network_mode: host` provides in this instance, I'd like to add that (ports and what not) manually so I can  continue using my network.

**Comment by marcelveldt** [comment, marked as answer]
because ports are dynamically opened/used and we rely on things like multicast for discovery.
There is an alternative to using host networking (which is even better imo) which is using a bridge/macvlan network. 

In any case, the MA container needs full L2 access to the network

**Comment by AlbertoKarakoutev** [comment]
Thanks, I'll keep at it then with a `network_mode: bridge`
